R/gh_request.R

```diff
@@ -133,16 +134,83 @@ gh_set_dest <- function(x) {
 #' <https://github.com/settings/tokens>.
 #'
 #' Currently it consults the `GITHUB_PAT` and `GITHUB_TOKEN`
-#' environment variables, in this order.
+#' environment variables, in this order. If none of these are set and
+#' the session is interactive, [gh_interactive_oauth_token()] is invoked
+#' to obtain an oauth token interactively.
 #'
 #' @return A string, with the token, or a zero length string scalar,
 #' if no token is available.
 #'
 #' @export
-
 gh_token <- function() {
   token <- Sys.getenv('GITHUB_PAT', "")
-  if (token == "") Sys.getenv("GITHUB_TOKEN", "") else token
+  if (token == "") token <- Sys.getenv("GITHUB_TOKEN", "")
+  if (token == "" && interactive()) {
+    gh_interactive_oauth_token()$credentials$access_token
```

Comment:
The problem with this, is that gh also works fine without any token, or even a GH account, and this forces you to get a token, right?

Reply:
Yeah it would be nice to think about a workflow where user either says "let's do the OAuth token thing" first. Or fails in a non-scary way and is prompted to "do the Oauth token thing". I.e. make this opt-in, but a very easy to opt-in.

Reply:
Ah yes. Good point! Let me look into a more optional/intelligence fallback. Thanks for the feedback.